kyuubi-server/src/main/scala/org/apache/kyuubi/engine/KubernetesApplicationOperation.scala

```diff
@@ -295,16 +314,64 @@ object KubernetesApplicationOperation extends Logging {
 
   def toLabel(tag: String): String = s"label: $LABEL_KYUUBI_UNIQUE_KEY=$tag"
 
-  def toApplicationState(state: String): ApplicationState = state match {
-    // https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L2396
-    // https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/
+  def toApplicationState(
+      pod: Pod,
+      appStateFromContainer: Boolean,
+      appStateContainer: String): ApplicationState = {
+    toApplicationStateAndError(pod, appStateFromContainer, appStateContainer)._1
+  }
+
+  def toApplicationStateAndError(
+      pod: Pod,
+      appStateFromContainer: Boolean,
+      appStateContainer: String): (ApplicationState, Option[String]) = {
+    val containerStateToBuildAppState = if (appStateFromContainer) {
+      pod.getStatus.getContainerStatuses.asScala.find(_.getState == appStateContainer).map(
+        _.getState)
```

Comment:
avoid breaking line after `func(\n lambda ...`

```suggestion
      pod.getStatus.getContainerStatuses.asScala
        .find(_.getState == appStateContainer).map(_.getState)
```